File: ProblemSet/Lists/typescript/deleteMiddle.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function deleteMiddle(head: ListNode | null): ListNode | null {
  let getListLength = (head: ListNode | null): number => {
    let length = 0;
    while (head) {
      length++;
      head = head.next;
    }
    return length;
  }

  if (!head) {
    return null;
  }

  let ptr: ListNode | null = head;
  let listLength = getListLength(ptr);

  if (listLength <= 1) {
    return null;
  }

  for (let i = 0; i < Math.floor(listLength / 2) - 1; i++) {
    ptr = ptr ? ptr.next : null;
  }

  if (ptr) {
    ptr.next = ptr?.next?.next ?? null;
    console.log(ptr.val)
  }
  return head;
};

File: ProblemSet/Lists/python/deleteMiddle.py
from typing import Optional
# Definition for singly-linked list.


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None or head.next is None:
            return None
        slowPtr = head
        fastPtr = head.next.next
        while fastPtr is not None and fastPtr.next is not None:
            slowPtr = slowPtr.next
            fastPtr = fastPtr.next.next
        slowPtr.next = slowPtr.next.next
        return head
